instrumentation/google.golang.org/grpc/otelgrpc/test/interceptor_test.go

```diff
@@ -583,24 +583,67 @@ func TestStreamClientInterceptorWithError(t *testing.T) {
 	assert.Equal(t, codes.Error, span.Status().Code)
 }
 
-func TestServerInterceptorError(t *testing.T) {
-	sr := tracetest.NewSpanRecorder()
-	tp := trace.NewTracerProvider(trace.WithSpanProcessor(sr))
-	usi := otelgrpc.UnaryServerInterceptor(otelgrpc.WithTracerProvider(tp))
-	deniedErr := status.Error(grpc_codes.PermissionDenied, "PERMISSION_DENIED_TEXT")
-	handler := func(_ context.Context, _ interface{}) (interface{}, error) {
-		return nil, deniedErr
+var serverChecks = []struct {
+	grpcCode     grpc_codes.Code
+	wantSpanCode codes.Code
+	name         string
+}{
+	{
+		grpcCode:     grpc_codes.OK,
+		wantSpanCode: codes.Unset,
+		name:         "ok",
+	},
+	{
+		grpcCode:     grpc_codes.Canceled,
+		wantSpanCode: codes.Unset,
+		name:         "cancelled",
+	},
+	{
+		grpcCode:     grpc_codes.NotFound,
+		wantSpanCode: codes.Unset,
+		name:         "not.found",
+	},
+	{
+		grpcCode:     grpc_codes.PermissionDenied,
+		wantSpanCode: codes.Unset,
+		name:         "permission.denied",
+	},
+	{
+		grpcCode:     grpc_codes.Unimplemented,
+		wantSpanCode: codes.Error,
+		name:         "unimplemented",
+	},
+	{
+		grpcCode:     grpc_codes.Internal,
+		wantSpanCode: codes.Error,
+		name:         "internal",
+	},
+	{
+		grpcCode:     grpc_codes.Unauthenticated,
+		wantSpanCode: codes.Unset,
+		name:         "unauthenticated",
+	},
+}
+
+func assertServerErr(t *testing.T, err error, grpcCode grpc_codes.Code) {
+	// the error should be nil only if the gRPC code is OK, otherwise it should be the same as the gRPC error
+	if grpcCode == grpc_codes.OK {
+		assert.NoError(t, err)
+	} else {
+		assert.Equal(t, grpcCode, status.Code(err))
```

Comment:
Can we add `grpcErr` to `serverChecks`?

So that it can be used here in the assertion by simply

```go
   assert.Equal(t, grpcCode, err)
```

and use the same error when setting up the handler:

```go
handler := func(_ context.Context, _ interface{}) (interface{}, error) {
	return nil, check.grpcErr
}
```

Reply:
I managed to remove this "if" branching and avoid duplicating it in `serverChecks` - cause `status.Error` returns nil if input code is "OK", so  the handler + the assert will use `nil` anyway in case of OK status